packages/gatsby/src/schema/node-model.js

```diff
@@ -253,6 +413,60 @@ class LocalNodeModel {
   }
 }
 
+class ContextualNodeModel {
```

Comment:
Node model that has a context, in tihs case it's basically just path. It forwards everything but `trackPageDependencies` to node model and adds default path otherwise.